lib/test-requirements.txt

```diff
@@ -10,12 +10,23 @@ opencv-python>=4.5.3
 plotly>=5.3.1
 pyspark>=3.1.1
 pydot>=1.4.2
-rich>=11.2.0
 scipy>=1.7.3
 seaborn>=0.11.2
 setuptools>=65.5.1
 watchdog>=2.1.5
 
+urllib3>=1.9
+
+hypothesis>=6.17.4
+parameterized
+pytest
+pytest-cov
+requests-mock
+testfixtures
+
+
+mypy-protobuf>=3.2
+
```

Comment:
Nit: I think the convention here is to alphabetize all dependencies in the list (and also no extra newlines).

Re: the new requirements themselves - it looks like most of these are duplicated from `dev-requirements.txt`. Should we just be installing `dev-requirements.txt` for this set of tests, rather than duplicating? Or does that break something with min-dependency testing?

(If the latter, can we add documentation to `test-requirements.txt` explaining the duplication?)

Reply:
Some dev requirements (mainly `black`) would impose much higher minimum versions of dependencies we share with them, which I wanted to avoid.

Reply:
Possibly some of these should be factored out into a 3rd requirements file

Reply:
It's also possible that we don't mind requiring a more recent typing extensions version, or could roll back to an older version of black that doesn't need something so recent (don't remember why we're on the version we have).